### DISCUSSION TITLE
t3api 1.2.1 and doctrine/cache version compatibility

### DISCUSSION THREAD

**Discussion by bandrachnik** [sourcebroker/t3api#37] [Q&A] [answered]
Hi,
 
If I install 1.2.1 composer will get version 2.0 for Doctrine Cache , which is not compatible with t3api,
Since the class SerializerService  uses Deprecated Class

https://github.com/doctrine/cache/blob/1.11.x/lib/Doctrine/Common/Cache/FilesystemCache.php

It work with version 1.11., but not with ^2

it used in CachedReader  
/**
 * A cache aware annotation reader.
 *
 * @deprecated the CachedReader is deprecated and will be removed
 *             in version 2.0.0 of doctrine/annotations. Please use the
 *             {@see \Doctrine\Common\Annotations\PsrCachedReader} instead.
 */

**Comment by klamparski** [comment]
Hi @bandrachnik .

Thank you for reporting this issue. I didn't have time yet to reproduce your issue but as I see it occurs on TYPO3 >= v10.4.16. The point is that since v10.4.16 TYPO3 allows to use `doctrine/dbal` >= 2.13.1 and `doctrine/dbal` >= 2.13.2 allows to use `doctrine/cache` >= 2.0.

Temporary solution I could suggest for you - please try to add `doctrine/cache: ^1.0` as your requirement in your project root composer.json. As I see it should not bring any conflicts in composer. Another solution, which would surely work but is not perfect is to downgrade TYPO3 to 10.4.15.

We will implement solution in t3api core as soon as possible.

If you would try suggested solution or implement another one please let us know if worked. Thanks in advance.

**Reply by bandrachnik** [reply, marked as answer]
it's work with:   require  "doctrine/cache": "1.11.3"
